safe/report/extractors/analysis_provenance_details.py

```diff
@@ -15,6 +20,209 @@
+        'provenances': headerize(
+            sorted_keywords_by_order(exposure_keywords, keywords_order))
+    }
+
+    # convert value if there is dict_keywords
+    provenances = exposure_provenance['provenances']
+    exposure_provenance['provenances'] = resolve_dict_keywords(provenances)
+
+    # aggregation keywords could be None so we don't define dict here
+    aggregation_keywords = impact_report.impact_function.provenance[
+        'aggregation_keywords']
+
+    header = resolve_from_dictionary(
+        provenance_format_args, 'aggregation_header')
+    provenance_format = resolve_from_dictionary(
+        provenance_format_args, 'aggregation_format')
+
+    aggregation_provenance = {
+        'header': header.title(),
+        'provenances': None
+    }
+
+    # only if aggregation layer used
+    if aggregation_keywords:
+        # we define dict here to create a different object of keyword
+        aggregation_keywords = dict(aggregation_keywords)
+
+        # aggregation_keywords doesn't have aggregation_layer path information
+        aggregation_layer = impact_report.impact_function.provenance.get(
+            'aggregation_layer')
+        aggregation_keywords['aggregation_layer'] = aggregation_layer
+
+        aggregation_provenance['provenances'] = headerize(
+            sorted_keywords_by_order(aggregation_keywords, keywords_order))
+
+        # convert value if there is dict_keywords
+        provenances = aggregation_provenance['provenances']
+        aggregation_provenance['provenances'] = resolve_dict_keywords(
+            provenances)
+
+    else:
+        aggregation_not_used = resolve_from_dictionary(
+            extra_args, ['defaults', 'aggregation_not_used'])
+        aggregation_provenance['provenances'] = aggregation_not_used
+
+    all_provenance_keywords = dict(impact_report.impact_function.provenance)
+
+    # we add debug mode information to the provenance
+    if debug_mode:
+        all_provenance_keywords['debug_mode'] = 'On'
+    else:
+        all_provenance_keywords['debug_mode'] = 'Off'
+
+    header = resolve_from_dictionary(
+        provenance_format_args, 'analysis_environment_header')
+    analysis_environment_provenance_items = OrderedDict()
+    analysis_environment_provenance_keys = [
+        'os',
+        'inasafe_version',
+        'debug_mode',
+        'qgis_version',
+        'qt_version',
+        'gdal_version',
+        'pyqt_version']
+
+    for item in analysis_environment_provenance_keys:
+        analysis_environment_provenance_items[item] = (
+            all_provenance_keywords[item])
+
+    analysis_environment_provenance = {
+        'header': header.title(),
+        'provenances': headerize(analysis_environment_provenance_items)
+    }
+
+    impact_function_name = impact_report.impact_function.name
+    header = resolve_from_dictionary(
+        provenance_format_args, 'impact_function_header')
+    provenance_format = resolve_from_dictionary(
```

Comment:
__[flake8]__
* [F841] local variable 'provenance_format' is assigned to but never used

<!-- Comment by [SideCI](https://sideci.com) -->